Python/ast_opt.c

```diff
@@ -29,6 +40,82 @@ typedef struct {
         --(ST)->recursion_depth; \
     } while(0)
 
+
+static ControlFlowInFinallyState
+overwrite_state(_PyASTOptimizeState *state, bool finally, bool funcdef, bool loop)
+{
+    ControlFlowInFinallyState saved = state->cf_finally;
+    state->cf_finally.in_finally = finally;
+    state->cf_finally.in_funcdef = funcdef;
+    state->cf_finally.in_loop = loop;
+    return saved;
+}
+
+static int
```

Comment:
```suggestion
static void
```
This can't fail, and the error checking complicates the code